src/excmds.ts

```diff
@@ -467,6 +469,26 @@ export function focusinput(nth: number|string) {
         // we can look up in future (tabindex, selector?), perhaps along with
         // a way to remember the page it was on?
     }
+    else if (nth === "-n") {
+        // attempt to find next input
+        let inputs = DOM.getElemsBySelector(INPUTTAGS_selectors,
+            [DOM.isSubstantial]) as HTMLElement[]
+        if (inputs.length) {
+            let index = inputs.indexOf(LAST_USED_INPUT) + 1
+            if (index >= inputs.length) index = 0
+            inputToFocus = <HTMLElement>inputs[index]
+        }
+    }
+    else if (nth === "-N") {
+        // attempt to find previous input
+        let inputs = DOM.getElemsBySelector(INPUTTAGS_selectors,
+            [DOM.isSubstantial]) as HTMLElement[]
+        if (inputs.length) {
+            let index = inputs.indexOf(LAST_USED_INPUT) - 1
+            if (index <= -1) index = inputs.length + index
+            inputToFocus = <HTMLElement>inputs[index]
```

Comment:
You shouldn't just copy and paste code like this: it obscures that two code paths are basically the same and increases the chance that bug fixes or other changes to one branch won't be applied to the other. Better to find the points of commonality and abstract or reuse.

In this case you can just use the mod function to replace this line: `if (index <= -1) index = inputs.length + index` and then the only difference is the sign in `let index = inputs.indexOf(LAST_USED_INPUT) - 1`.

The minimal change would be something like this:

```ts
    else if (nth === "-n" || nth === "-N") {
        // attempt to find next input
        let inputs = DOM.getElemsBySelector(INPUTTAGS_selectors,
            [DOM.isSubstantial]) as HTMLElement[]
        if (inputs.length) {
            let index = inputs.indexOf(LAST_USED_INPUT)
            if (nth === "-n") index++
            else index--
            index = index.mod(inputs.length)
            inputToFocus = <HTMLElement>inputs[index]
        }
    }
```